Lib/test/test_capi/test_object.py

```diff
@@ -247,5 +247,32 @@ def func(x):
 
         func(object())
 
+    def test_object_setattr_null_exc(self):
```

Comment:
I suggest to move this test to `Lib/test/test_capi/test_abstract.py`, where all other tests for `PyObject_SetAttr` are.